dwave/optimization/include/dwave-optimization/array.hpp

```diff
@@ -696,25 +698,46 @@ class Array {
         // not obvious what the value to the user would be. If we ever need them
         // they are easy to add.
      public:
+        /// A std::random_access_iterator over the values.
         using iterator = ArrayIteratorImpl_<false>;
+        /// A std::random_access_iterator over the values.
         using const_iterator = ArrayIteratorImpl_<true>;
 
+        /// Create an empty view.
         View() = default;
 
+        /// Create a view from an Array and a State.
         View(const Array* array_ptr, const State* state_ptr)
                 : array_ptr_(array_ptr), state_ptr_(state_ptr) {
             assert(array_ptr && "array_ptr must not be nullptr");
             assert(state_ptr && "state_ptr must not be nullptr");
         }
 
+        /// Return a reference to the nth element of the view.
         const double& operator[](ssize_t n) const;
 
+        /// Return a reference to the nth element of the view.
+        ///
+        /// This function checks whether n is within bounds and throws a
```

Comment:
```suggestion
        /// This function checks whether ``n`` is within bounds and throws a
```

You use the backticks below in line 774 (Exactly ``sizeof(double)``.) and it prevents a plain "n" from looking like a typo.

Reply:
Looks like Doxygen actually wants `This function checks whether @p n is within bounds...`. See https://www.doxygen.nl/manual/commands.html#cmdp. How's that for readability :roll_eyes:

Reply:
Ah, Doxygen has markdown support, but it wants single backticks for code. See https://www.doxygen.nl/manual/markdown.html#md_codespan

Reply:
Rivalled only by "use <tt>multiple words</tt>"

Reply:
great, then also line 774